src/main/java/com/thealgorithms/dynamicprogramming/LongestStringChain.java

```diff
@@ -0,0 +1,64 @@
+/** Author : Siddhant Swarup Mallick
+ * Github : https://github.com/siddhant2002
+ */
+
+/** Program Description - To find the length of the longest string chain*/
+/** Program Description link - https://medium.com/nerd-for-tech/longest-string-chain-daily-challenge-may-a07eda8b112b */
+
+/** Wikipedia Link - https://en.wikipedia.org/wiki/Longest_common_substring */
```

Comment:
It's not the longest common substring problem, please update the link if this is a well-known problem, we only add well-known algorithms